jobs/buildhub/to_kinto.py

```diff
@@ -0,0 +1,173 @@
+"""
+Read records as JSON from stdin, and pushes them on a Kinto server concurrently.
+
+Usage:
+
+    $ echo '{"data": {"title": "a"}}
+    {"data": {"title": "b"}}
+    {"data": {"title": "c"}}' | to-kinto --server=https://... --bucket=bid --collection=cid --auth=user:pass
+
+It is meant to be combined with other commands that output records to stdout :)
+
+    $ load-inventory filename.csv | to-kinto --auth=user:pass
+    $ scrape-archives | to-kinto --auth=user:pass
+
+"""
+import asyncio
+import async_timeout
+import concurrent.futures
+import json
+import logging
+import sys
+
+from kinto_http import cli_utils
+
+
+DEFAULT_SERVER = "http://localhost:8888/v1"
+DEFAULT_BUCKET = "default"
+DEFAULT_COLLECTION = "cid"
+NB_THREADS = 3
+NB_RETRY_REQUEST = 3
+WAIT_TIMEOUT = 5
+
+logger = logging.getLogger(__name__)
+
+done = object()
+
+
+def publish_records(client, records):
+    """Synchronuous function that pushes records on Kinto in batch.
+    """
+    with client.batch() as batch:
+        for record in records:
+            if "id" in record["data"]:
+                batch.update_record(**record)
+            else:
+                batch.create_record(**record)
+    results = batch.results()
+
+    # Batch don't fail with 4XX errors. Make sure we output a comprehensive
+    # error here when we encounter them.
```

Comment:
I have having to do this here. The kinto client should offer an option `with client.batch(silent=False)`